hashring/hashring.go

```diff
@@ -76,6 +80,7 @@ func New(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
 		hashfunc: func(str string) int {
 			return int(hashfunc([]byte(str)))
 		},
+		logger: logging.Logger("ring"),
```

Comment:
I like the idea of having the local in every log message from ringpop, because it makes it easy to reason about which member logged a line. But maybe we should move that logic into 1 place instead of passing that field in with every module logger. It also makes it transparent for all the modules in ringpop instead that they all call the WithField on their own logger.